index.js

```diff
@@ -0,0 +1,32 @@
+import {
+  createConnection,
+  ProposedFeatures,
+  TextDocuments
+} from 'vscode-languageserver/node.js'
+import {TextDocument} from 'vscode-languageserver-textdocument'
+import {configureUnifiedLanguageServer} from './server.js'
+
+/**
+ * @typedef {Pick<
+ *   import('unified-engine').Options,
+ *   'ignoreName' | 'packageField' | 'pluginPrefix' | 'plugins' | 'rcName'
+ * > & {
+ *   defaultSource?: string
+ * }} Options
+ */
```

Comment:
I’d say define types as close to where they are used (so `server.js`, which includes `configureUnifiedLanguageServer`). Any reason you prefer it here?

Reply:
This way only the types of the public interface need to be published. When this is moved into `server.js`, combined with how typedefs are used, 2 additional dependencies are needed: `vfile-message` and `@types/unist`, even though they aren’t important to users.

These packages are already pulled in as indirect dependencies, so it’s not a big deal, but I personally prefer not having to add otherwise unneeded dependencies.

Reply:
You mean to ignore internal `*.d.ts` files when publishing, only publishing `index.d.ts`?
Is there a benefit of that, other than publishing a few files less?
Because the drawback to me is that it becomes hard to figure out if the types are correct or not locally before publishing, and quite easy to acceidentally mess up?

Reply:
It doesn't just save a few files, but also dependencies.

If that tradeoff is ok with you, I'll refactor.

Reply:
I find it rather weird to define types in the main entry point and require those types from deeper files. As this “flow” is exactly the other way of how JS code is used: defined in deep files, exported higher up.
For bigger projects, as you say, that means including type dependencies in `index.js` in `package.json`, but not when they occur in other files: really easy to mess up or forget about.

The rule of: include every `import`, whether code or type, in `dependencies`, seems the simplest to me. Especially because, well, the dependencies we’re talking about: `unified-engine`, `vfile-message`, `unist`, `vscode-language-server`, those will already be included in `node_modules`?